### PR TITLE
Porting python implementation to ruby

### PR DESCRIPTION
**Pull request by ProGM**
Hi there,

This is a porting of the python implementation in Ruby. I need this library for our company projects in Ruby on Rails 💎 

I hope you'll decide to merge this ;)

Implementation details:
* Supports ruby 2.7+
* It's [ready to be released on rubygems.org](https://github.com/monade/flexible-polyline/blob/feature/ruby-implementation/ruby/flexpolyline.gemspec)
* Tests use made with rspec, implementing the same suite of Python (runnable via `bundle exec rspec`)
* Documentation using Yard (runnable via `bundle exec yardoc`)
* It's configured with rubocop for code linting

### CODE REVIEW

**Comment by sergykorets** [comment]
Please merge

File: ruby/spec/spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+require 'rspec'
+require 'flexpolyline'
+
+RSpec::Expectations.configuration.warn_about_potential_false_positives = false
+
+# Dir[File.expand_path('support/*.rb', __dir__)].each { |f| require f }
```
Comment (roshats) [inline ruby/spec/spec_helper.rb:8]: This code is not needed. Please remove it.

File: ruby/flexpolyline.gemspec [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  s.date        = '2022-08-26'
+  s.summary     = 'Flexible Polyline encoding: a lossy compressed representation of a list of coordinate pairs or triples.'
+  s.description = 'Flexible Polyline encoding: a lossy compressed representation of a list of coordinate pairs or triples.'
+  s.authors     = ['HERE Europe B.V.', 'Mònade srl']
+  s.files = Dir['lib/**/*']
+  s.test_files = Dir['spec/**/*']
+  s.required_ruby_version = '>= 2.7.0'
+  s.homepage    = 'https://rubygems.org/gems/flexpolyline'
+  s.license     = 'MIT'
+  s.add_dependency 'activesupport', ['>= 5', '< 8']
```
Comment (roshats) [inline ruby/flexpolyline.gemspec:18]: Please do not depend on `activesupport`.

**Review by roshats — CHANGES_REQUESTED** [review]

**Comment by ProGM** [comment]
@roshats Hi, I've made all the requested changes. Let me know if it's all fine or it's ready to be merged :)

**Comment by roshats** [comment]
Thanks @ProGM .

@VeaaC , I think me can merge this Pull Request and publish ruby gem.

**Review by roshats — APPROVED** [review]

**Comment by VeaaC** [comment]
@ProGM the commits needs to be signed, then I can go ahead an merge ( https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification )

**Comment by ProGM** [comment]
@VeaaC @roshats Commits signed ;)

**Comment by VeaaC** [comment]
Merging. @roshats  / @ProGM  can you also add some CI to test against the shared conversion tests in https://github.com/heremaps/flexible-polyline/tree/master/test ?

**Pull request merged**